### PR TITLE
feat: parallel processing

### PR DESCRIPTION
**Pull request by ctmm**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -19,12 +21,27 @@ module.exports = {
...
         // The module of your components (required), eg "@ionic/core".
+        // Add multiple modules by using an array eg ["@ionic/core", "my-stencil-components"]
         module: "your-stencil-components-package",
 
         // Stencil renderToString options (optional): https://stenciljs.com/docs/hydrate-app#configuration-options
         renderToStringOptions: {
           prettyHtml: true,
+          removeHtmlComments: true,
+          // Must be a string as it's "evaled" by the plugin
+          afterHydrate: `(document) => document.querySelectorAll('style').forEach((tag) => tag.parentElement.removeChild(tag));`,
```
Comment (nicow-vi) [inline README.md]: just curious, why get the parent and removeChild and not just tag.remove()?

File: src/hydrateFiles.js [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+  files,
+  numWorkers,
+  pkg,
+  renderToStringOptions,
+  reporter
+) => {
+  //divide work among workers
+  const batchSize = Math.floor(files.length / numWorkers);
+  const workerPromises = [];
+  for (var i = 0; i < numWorkers; i++) {
```
Comment (nicow-vi) [inline src/hydrateFiles.js:56]: could be a let :p

File: src/hydrateFiles.js [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+ * Loops over all files passed in, sets up initial config and
+ * passes through hyrdate options and stencil pkg to the handlers.
+ *
+ * @param {*} files
+ * @param {*} pkg
+ * @param {*} reporter
+ * @param {*} config
+ */
+exports.hydrateFiles = async (files, pkg, reporter, config) => {
+  const os = require("os");
```
Comment (nicow-vi) [inline src/hydrateFiles.js:84]: would move import to the top of the file

File: gatsby-node.js [hunk 4]
```diff
@@ -1,68 +1,28 @@
...
-          return;
-        }
-
-        throw e;
-      }
-    })
-  );
-};
\ No newline at end of file
+  const packages = Array.isArray(module) ? module : [module]
```
Comment (demirgazetic) [inline gatsby-node.js]: NIT:

Can this be written as [].concat(module)?

File: gatsby-node.js [hunk 5]
```diff
@@ -1,68 +1,28 @@
...
+
+  // Could still experience mem issues with this but would require a lot of files
+  const files = glob.sync("public/**/*.html", { nodir: true });
+  console.info("Stencil hydration starting");
+  for (let i = 0; i < packages.length; i++) {
+    if (files.length > 0) {
+      console.log(`${files.length} files found to hydrate.`);
+      await hydrateFiles(files, packages[i], reporter, pluginOptions);
+    } else {
+      console.error("No files found to hydrate");
```
Comment (demirgazetic) [inline gatsby-node.js]: NIT:

This is more of a console.warn then an error?

**Review by nicow-vi — APPROVED** [review]

**Review by demirgazetic — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (ctmm) [reply README.md]: Ignorance 🥲

**Pull request merged**